kombu/transport/redis.py

```diff
@@ -1079,6 +1079,11 @@ def _purge(self, queue):
 
     def close(self):
         self._closing = True
+        if self._in_poll:
```

Comment:
can we try some integration tests for the changes as well? https://github.com/celery/kombu/blob/main/t/integration/test_redis.py